src/filter.c

```diff
@@ -279,3 +280,60 @@ void quisk_ccfFilter(complex float * inSamples, complex float * outSamples, int
     }
 }
 
+void fir_quick_init(struct fir_quick** filter, float* coeffs, int numCoefficients)
+{
+    assert(filter != NULL);
+    
+    *filter = MALLOC(sizeof(struct fir_quick));
+    assert(*filter != NULL);
+    
+    (*filter)->lookbackBuffer = MALLOC(sizeof(complex float) * numCoefficients);
+    assert((*filter)->lookbackBuffer != NULL);
+    memset((*filter)->lookbackBuffer, 0, sizeof(complex float) * numCoefficients);
+    
+    (*filter)->coefficients = coeffs;    
+    (*filter)->numCoefficients = numCoefficients;
+    (*filter)->currentLookbackWritePosition = 0;
+}
+
+void fir_quick_destroy(struct fir_quick** filter)
+{
+    assert(filter != NULL);
+    assert(*filter != NULL);
+    
+    FREE((*filter)->lookbackBuffer);
+    FREE(*filter);
+    *filter = NULL;
+}
+
```

Comment:
Do we really need new init functions?  Could the original ones be use?  Perhaps this saves some memory.

Reply:
I'm not fully sure we can remove the original ones at the moment. See below.